### PR TITLE
Add session.previous_id

### PR DESCRIPTION
**Pull request by breedx-splk**
## Changes

Introduces a convention to represent the previous session id for client instrumentation. This can be used by telemetry backends to create links from the old/expired session to the new one.

Existing art exists in Android via the [SessionIdChangeTracer](https://github.com/open-telemetry/opentelemetry-android/blob/main/instrumentation/src/main/java/io/opentelemetry/android/SessionIdChangeTracer.java#L21). 

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Review by martinkuba — APPROVED** [review]

File: docs/general/session.md
Comment (scheler) [inline docs/general/session.md:24]: Not sure if it's overthinking but do we want to encode both in one id?

For eg., session.id could be an 32 byte identifier, with 16 bytes of it for the previous session. So, a totally new session's id would start with 16 zeroes at the beginning.

**Review by LikeTheSalad — APPROVED** [review]

File: docs/general/session.md
Comment (LikeTheSalad) [reply docs/general/session.md:24]: I think this could cause more problems and I'm not sure if the benefits would be worth it, if I understood correctly we'd have to constrain the format of the `session.id` to always be 32 bytes long, all numbers and we'd always have to make it "previous_id"-aware by having to prepend 16 zeroes to it all the time, even for the cases where knowing about the previous id might not be a requirement, we might also have to create apis to set both ids separately and the sdks would have to merge them later, otherwise it could be a bit error-prone for anybody to manually build the right format of the id as a single attr. From the server side of things, it'd always have to process incoming session ids to split them up and make them easier to read later I believe.

**Comment by AlexanderWert** [comment]
I think I get the idea of this proposal. But I think it would help if we had mobile- or client-specific semantic conventions that would describe how these attributes are intended to be used. I think it's clear for `session.id` (i.e. it would be just an attribute that's being added to all mobile/client signals/events, except for metrics due to cardinality). 

But, what about `session.previous_id`? Would that be set on every signal/event as well (e.g. every mobile span would always have `session.id` and `session.previous_id`)? Or, is the intention rather to send a specific event that would indicate a "session transition" from old session to new session that would contain those two attributes?

I think documenting this reasoning as mobile semantic conventions would be great!

**Comment by breedx-splk** [comment]
> But, what about `session.previous_id`? Would that be set on every signal/event as well (e.g. every mobile span would always have `session.id` and `session.previous_id`)? Or, is the intention rather to send a specific event that would indicate a "session transition" from old session to new session that would contain those two attributes?

Yeah, I linked to the tracer in android to demonstrate, but yeah, the idea is to send a session change event. I can add some text in this same file to give more of an explanation of intent. Is this the right place for that, or, because it's behavioral does it belong in the spec repo?

**Comment by AlexanderWert** [comment]
> Yeah, I linked to the tracer in android to demonstrate, but yeah, the idea is to send a session change event. I can add some text in this same file to give more of an explanation of intent. Is this the right place for that, or, because it's behavioral does it belong in the spec repo?

I think a new area of semantic conventions under `/docs/mobile/` would make sense (in addition to this PR, where just the session is being defined). In `/docs/mobile/` we then can describe all the mobile-specific conventions such as types of mobile spans and their attributes to capture, description of mobile events to send, etc.

With https://github.com/open-telemetry/semantic-conventions/issues/197 the definition of the session attributes would go into the registry anyways and the mobile semantic conventions would just reference it.

File: docs/general/session.md
Comment (scheler) [reply docs/general/session.md:24]: @breedx-splk this comment can be resolved. thanks.

**Review by bryce-b — APPROVED** [review]

**Review by scheler — APPROVED** [review]

**Review by AlexanderWert — APPROVED** [review]

**Comment by AlexanderWert** [comment]
@open-telemetry/semconv-mobile-approvers Can we get more reviews on this one please?

**Review by surbhiia — APPROVED** [review]

**Comment by breedx-splk** [comment]
Hi @open-telemetry/specs-semconv-maintainers -- we have 4 approvals on this. Any reason to hold this up? 🙏🏻

**Pull request merged**
